subprojects/jvm-services/src/main/java/org/gradle/jvm/toolchain/internal/MavenToolchainsInstallationSupplier.java

```diff
@@ -85,7 +88,22 @@ protected Set<InstallationLocation> findCandidates() {
                 for (int i = 0; i < nodes.getLength(); i++) {
                     Node item = nodes.item(i);
                     if (item != null && item.getNodeType() == Node.TEXT_NODE) {
-                        locations.add(item.getNodeValue().trim());
+                        String nodeValue = item.getNodeValue().trim();
+                        Matcher matcher = ENV_VARIABLE_PATTERN.matcher(nodeValue);
+                        StringBuffer resolvedValue = new StringBuffer();
+                        if (matcher.find()) {
+                            String envVariableName = matcher.group(1);
+                            Provider<String> envVariableValue = getEnvironmentProperty(envVariableName);
+                            if (envVariableValue == null || envVariableValue.getOrNull() == null) {
```

Comment:
I have made it this way at first but somehow envVariableValue is returned as null object and gave me null exception in the tests. Maybe need some investigation as this should not happen

Reply:
Ok, I'll follow up on that, let's ignore in this context

Reply:
I think I know what the issue is: usage of `Mock` in tests.
If you do not use something like `Providers.ofNullable(null)` for the unknown env val, then you get a null indeed. But it is a special case because of the tests.

❓ Can you verify this?

Reply:
That helped!
I have tried it and removed the null check and it worked successfully.